frontend/server/src/main/java/org/pytorch/serve/wlm/BatchAggregator.java

```diff
@@ -63,32 +64,34 @@ public void sendResponse(ModelWorkerResponse message) {
                 // this is from initial load.
                 return;
             }
+            Iterator<Predictions> predictionsIterator = message.getPredictions().iterator();
 
-            for (Predictions prediction : message.getPredictions()) {
+            while (predictionsIterator.hasNext()) {
+                Predictions prediction = predictionsIterator.next();
                 String jobId = prediction.getRequestId();
-                Job job = jobs.remove(jobId);
+                Job job = jobs.get(jobId);
+
                 if (job == null) {
                     throw new IllegalStateException("Unexpected job: " + jobId);
                 }
                 job.response(
-                        prediction.getResp(),
-                        prediction.getContentType(),
-                        prediction.getStatusCode(),
-                        prediction.getReasonPhrase(),
-                        prediction.getHeaders());
+                    prediction.getResp(),
+                    prediction.getContentType(),
+                    prediction.getStatusCode(),
+                    prediction.getReasonPhrase(),
+                    prediction.getHeaders());
             }
+            
         } else {
-            for (String reqId : jobs.keySet()) {
-                Job j = jobs.remove(reqId);
-                if (j == null) {
-                    throw new IllegalStateException("Unexpected job: " + reqId);
+            for (Map.Entry<String, Job> j : jobs.entrySet()) {
+                
+                if (j.getValue() == null) {
+                    throw new IllegalStateException("Unexpected job: " + j.getKey());
```

Comment:
It is better to update the exception message to differentiate this exception for failure prediction with the exception at line 71.

Reply:
Fixed